app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -43,6 +44,14 @@ class UserViewModel(
   // Keep track of whether the user has connection to the internet or not.
   private val _hasConnection = MutableStateFlow(true)
   val hasConnection: StateFlow<Boolean> = _hasConnection
+
+  // Combine the connectivity status and the mode selected to decide if we show ui elements that
+  // require internet.
+  // (i.e the search bars)
+  val displayOnlineElementFlow =
+      combine(_isOnlineMode, _hasConnection) { isOnlineMode, hasConnection ->
+        isOnlineMode && hasConnection
+      }
```

Comment:
You could call distinctUntilChanged() to only emit when the value changes to avoid redundant UI updates.